inst/include/common/fims_vector.hpp

```diff
@@ -0,0 +1,437 @@
+     * not be called explicitly in FIMS, they may be required to run other std library functions.
+    */
+    
+    /**
+     * @brief Returns a reference to the element at specified location pos. No bounds checking is performed.
+     */
+    inline Type& operator[](size_t pos){
+        return this->vec_m[pos];
+    }
+    
+    /**
+     * @brief Returns a constant  reference to the element at specified location pos. No bounds checking is performed.
+     */
+    inline const Type& operator[](size_t n) const{
+        return this->vec_m[n];
+    }
+    
+    
+    /**
+     * @brief Returns a reference to the element at specified location pos. Bounds checking is performed.
+     */
+    inline Type& at(size_t n) {
+        return this->vec_m.at(n);
+    }
+    
+    /**
+     * @brief Returns a constant reference to the element at specified location pos. Bounds checking is performed.
+     */
+    inline const Type& at(size_t n) const{
+        return this->vec_m.at(n);
+    }
+    
+    /**
+     * @brief  Returns a reference to the first element in the container.
+     */
+    reference front() {
+        return this->vec_m.front();
+    }
+    
+    /**
+     * @brief  Returns a constant reference to the first element in the container.
+     */
+    const_reference front() const {
+        return this->vec_m.front();
+    }
+    
+    /**
+     * @brief  Returns a reference to the last element in the container.
+     */
+    reference back() {
+        return this->vec_m.back();
+    }
+    
+    
+    /**
+     * @brief  Returns a constant reference to the last element in the container.
+     */
+    const_reference back() const {
+        return this->vec_m.back();
+    }
+    
+    
+    /**
+     * @brief Returns a pointer to the underlying data array.
+     */
+    pointer data() {
+        return this->vec_m.data();
+    }
+    
+    
+    /**
+     * @brief Returns a constant pointer to the underlying data array.
+     */
+    const_pointer data() const {
+        return this->vec_m.data();
+    }
+    
+    //iterators
+    
+    
+    /**
+     * @brief Returns an iterator to the first element of the vector.
+     */
+    iterator begin() {
+        return this->vec_m.begin();
+    }
+    
+    /**
+     * @brief Returns an iterator to the element following the last element of the vector.
+     */
+    iterator end() {
+        return this->vec_m.end();
+    }
+    
+    
+    /**
+     * @brief Returns an constant iterator to the first element of the vector.
+     */
+    const_iterator begin() const {
+        return this->vec_m.begin();
+    }
+    
+    /**
+     * @brief Returns an const iterator to the element following the last element of the vector.
+     */
+    const_iterator end() const {
+        return this->vec_m.end();
+    }
+    
+    /**
+     * @brief Returns a reverse iterator to the first element of the reversed vector. It corresponds to the last element of the non-reversed vector.
```

Comment:
knit: The briefs on the following four functions related to reverse_iterator are hard to understand. If these are just copoed over from std::vector, that is probably fine but it might be helpful to add a big picture sentence about what these are doing for non-C++ experts